### PR TITLE
ROCIC-20 Batch conversion

### PR DESCRIPTION
**Pull request by davidlee-zuhlke**
Major changes:

- Added logic to detect if the input_path is a directory or file. if it's a directory, apply the specified conversion to all file in that directory
- Added a util function to filter files in the specified directory by extension

### CODE REVIEW

File: imgconvert.py [hunk 1]
```diff
@@ -12,13 +13,37 @@ def convert_roche_image(conversion_args):
...
 
     if not converter_func:
-        print(f"Conversion type {conversion_args.conversion_type} not found.")
+        print(f"Conversion type {type} not found.")
         return
 
-    converter_func(conversion_args.input_path, conversion_args.output_dir)
+    src_path = conversion_args.input_path
+
+    if(not os.path.exists(src_path)):
```
Comment (huynguyen-zuhlke) [inline imgconvert.py:31]: do not need outside parenthesis. 

**if not os.path.exists(src_path):**

File: src/utils.py [hunk 2]
```diff
@@ -18,3 +18,15 @@ def extract_datetime_from_keyframe(keyframe):
...
     return None
+
+
+def filter_files(dir, extension=None):
+    if(not os.path.isdir(dir)):
+        raise Exception(f"{dir} is not a directory")
+
+    matched = []
+    for file in os.listdir(dir):
+        if extension is None or file.endswith(extension):
```
Comment (huynguyen-zuhlke) [inline src/utils.py:29]: maybe we can check that only files are included in the dir, not any directories (dir may contain nested dir), using 

 **if os.path.isfile(os.path.join(directory, file)) and (extension is None or file.endswith(extension))**

File: imgconvert.py [hunk 3]
```diff
@@ -12,13 +13,37 @@ def convert_roche_image(conversion_args):
...
+        raise Exception(f"Input path does not exist")
+
+    if os.path.isdir(src_path):
+        ext = source_extension_dict.get(type)
+        source_images = filter_files(src_path, ext)
+
+        print(f"\nPerforming {type} conversion for images in {src_path}")
```
Comment (huynguyen-zuhlke) [inline imgconvert.py:38]: maybe we can check if the source_images is empty

File: imgconvert.py (hunk 1 above)
Comment (davidlee-zuhlke) [reply imgconvert.py]: Godo catch

File: imgconvert.py (hunk 3 above)
Comment (davidlee-zuhlke) [reply imgconvert.py]: Good idea

File: src/utils.py (hunk 2 above)
Comment (davidlee-zuhlke) [reply src/utils.py]: Yah that's another condition that needs to be checked

**Review by huynguyen-zuhlke — APPROVED** [review]

**Pull request merged**